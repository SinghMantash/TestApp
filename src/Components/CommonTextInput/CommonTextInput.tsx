import {StyleSheet, Text, TextInput, View} from 'react-native';
import React from 'react';
import {scaleHeight} from '../../styles/dimensions';
import colors from '../../styles/colors';
interface Props {
  placeholder: string;
  value: string;
  onChangeText?: any;
  validationError?: string;
  secureTextEntry?: boolean;
  editable: boolean;
  showHeading: boolean;
}
const CommonTextInput = ({
  placeholder,
  value,
  onChangeText,
  validationError,
  secureTextEntry,
  editable = true,
  showHeading,
}: Props) => {
  return (
    <View>
      {showHeading && <Text>{placeholder}</Text>}
      <TextInput
        placeholder={placeholder}
        style={styles.input}
        value={value}
        onChangeText={onChangeText}
        secureTextEntry={secureTextEntry}
        editable={editable}
      />
      <Text style={styles.errorText}>
        {!!validationError && validationError}
      </Text>
    </View>
  );
};

export default CommonTextInput;

const styles = StyleSheet.create({
  input: {
    marginTop: scaleHeight(10),
    borderWidth:1,
    borderColor:colors.secondaryColor,
    borderRadius:50,
    paddingLeft:16
  },
  errorText: {
    color: colors.error,
  },
});
