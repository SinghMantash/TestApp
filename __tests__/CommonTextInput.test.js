import 'react-native';
import React from 'react';

// Note: import explicitly to use the types shipped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import CommonTextInput from '../src/Components/CommonTextInput/CommonTextInput';
import {fireEvent, render, screen} from '@testing-library/react-native';
import {
  toBeEmptyElement,
  toHaveTextContent,
} from '@testing-library/jest-native';

expect.extend({toBeEmptyElement, toHaveTextContent});

const onChangeText = jest.fn();
const validationError = 'Test Error';
const userInput = 'Test';

it('CommonTextInput Snapshot Matching', () => {
  const tree = renderer
    .create(
      <CommonTextInput
        placeholder=""
        value=""
        onChangeText={() => {}}
        validationError=""
        secureTextEntry={false}
        showHeading={false}
        editable={false}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it('CommonTextInput editable Prop working ', async () => {
  render(
    <CommonTextInput
      testID="test-input"
      placeholder=""
      value=""
      onChangeText={onChangeText}
      validationError=""
      secureTextEntry={false}
      showHeading={false}
      editable={false}
    />,
  );
  fireEvent.changeText(screen.getByTestId('test-input'), userInput + '2');
  expect(onChangeText).not.toHaveBeenCalled();
});
it('CommonTextInput onChangeText Prop working ', async () => {
  render(
    <CommonTextInput
      testID="test-input"
      placeholder=""
      value=""
      onChangeText={onChangeText}
      validationError=""
      secureTextEntry={false}
      showHeading={false}
      editable={true}
    />,
  );
  fireEvent.changeText(screen.getByTestId('test-input'), userInput + '1');
  expect(onChangeText).toHaveBeenCalled();
});
it('CommonTextInput validationError Prop working ', async () => {
  render(
    <CommonTextInput
      testID="test-input"
      placeholder=""
      value=""
      onChangeText={onChangeText}
      validationError={validationError}
      secureTextEntry={false}
      showHeading={false}
      editable={true}
    />,
  );
  const err = await screen.findByTestId('test-input-error');
  expect(err).toHaveTextContent(validationError);
});
