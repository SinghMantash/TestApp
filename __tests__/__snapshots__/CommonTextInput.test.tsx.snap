// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <TextInput
    editable={false}
    onChangeText={[Function]}
    placeholder=""
    secureTextEntry={false}
    style={
      {
        "borderColor": "#F27900",
        "borderRadius": 50,
        "borderWidth": 1,
        "marginTop": 16.428571428571427,
        "paddingLeft": 16,
      }
    }
    value=""
  />
  <Text
    style={
      {
        "color": "red",
      }
    }
  />
</View>
`;
